### DISCUSSION TITLE
GPM is stable - any news regarding GQL support?

### DISCUSSION THREAD

**Discussion by annainfo** [terminusdb/terminusdb#1662] [Q&A] [answered]
https://github.com/orgs/terminusdb/projects/2#card-70200595 "A Bit Later" - "GQL Support"

https://github.com/orgs/terminusdb/projects/2#card-70200176 "Exploring" - "GraphQL Support" - but this could already be under "Done" according to 2022-11-15 https://terminusdb.com/blog/10-1-8-terminusdb-with-graphql/  "TerminusDB 10.1.8 has been released which means TerminusDB now comes with GraphQL to improve the developer experience."

So the page https://github.com/orgs/terminusdb/projects/2 is not up-to-date.

https://www.gqlstandards.org/ July, 2022 – GQL Status Update
"SQL/PGQ is interesting to people focused on GQL because the Graph Pattern Matching (GPM) language in SQL/PGQ is essentially identical to GPM in GQL. Since the technical work on SQL/PGQ is now stable, GPM in GQL is now stable."

TerminusDB could start working on support of GPM. And possibly be ready when the standard is published
https://ldbcouncil.org/event/fifteenth-tuc-meeting/attachments/keith-hare-property-graph-standards-process-and-timing.pdf

"Expected Dates
• DIS Ballot completes – December 2022
• Published standard – June 2023"

**Comment by luke-feeney** [comment, marked as answer]
Good point on GraphQL - we have updated the card as support is here. @annainfo 
 
Basically we don't have the capacity at the moment to extend to GQL. We'd really like to see it soon, but either somebody will have to take it on as a project or wait for us to get some time to develop. It would be cool to be one of the first graph engines with support. We think we need to build a combinator library in rust. This will facilitate query in GraphQL, utility queries so we can move some of the startup out of prolog, and then eventually could be used to implement GQL.

**Comment by annainfo** [comment]
Cf. https://github.com/orgs/terminusdb/discussions/2088 "Support ISO GQL:2024 : avoid vendor lock-in, facilitate benchmarking/comparability"
